File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def rec(x,n):
            if n == 0:
                return 1
            if n == 1:
                return x

            res = rec(x*x, n//2)
            return res if n%2 == 0 else res*x

        res = rec(x,abs(n))
        return res if n>=0 else 1/res 

File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        vector<int> res(temperatures.size(), 0);

        stack<pair<int,int>> stack;

        for (int i = 0 ; i < temperatures.size() ; i++){
            int t = temperatures[i];
            while ( !stack.empty() && t > stack.top().first){
                auto pair = stack.top();
                res[pair.second] = i - pair.second;
                stack.pop(); 
            }

            stack.push({t,i});
        }

        return res;
    }
};
